modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/DefaultExecutionMediator.java

```diff
@@ -146,7 +150,11 @@ private OperationExecutionResult executeWithInterceptors(OperationExecutor execu
       after(context, result, interceptors);
     }
 
-    return new OperationExecutionResult(result, exception, Optional.ofNullable(retryRequestHolder.get()));
+    Mono<Object> reactiveResult = result instanceof Publisher
+        ? from((Publisher) result)
+        : justOrEmpty(result);
```

Comment:
`concatMap` will drop empty mono's, so you'll need to handle this in OperationMessageProcessor if you don't want that and return the original event instead.